File: ksp/debilsort2.cpp
#include <iostream>
#include <vector>
#include <cstdio> 
#include <algorithm>
#include <set>
#include <string>
#include <map>
#include <queue>
#include <iomanip>
#include <cmath>

using namespace std;

#define REP(i, to) for(int i=0; i<to; i++)

typedef long long int LLI;
typedef pair<int, int> PII; 

LLI T(int n){
  if(n<=1) return 1;
  LLI sum=2*n;
  REP(i, n-1) sum += T(i+1); 
  return sum; 
}

#define MOD 1000000000

int main(){
  set<LLI> W;
  LLI x = 5;
  map<LLI, int> M; 
  map<int, LLI> Mb;
  int i=2; 
  while(W.find(x) == W.end()){
    W.insert(x); 
    Mb[i]=x;
    M[x]=i++; 
    x = (x*2 + 2) % MOD; 
  }
  cout << x << endl;
  cout << W.size() + 1 << "/" << M[x] << endl; 
  cout << Mb[1562513] << endl;
  cout << Mb[1562512] << endl;
  cout << Mb[1562511] << endl;
  cout << Mb[1562510] << endl;
  cout << Mb[1562509] << endl;
  cout << Mb[1562508] << endl;
  cout << Mb[1562507] << endl;
  cout << Mb[1562506] << endl;
  cout << Mb[1562505] << endl;
  
  return 0; 
}

File: ksp/2016-skolske/debilsort2.py
a = int(input())
mod = 1000000000

if a < 30:
  if a == 1:
    print(str(1))
  else:
    x = 5
    i = 2
    while i < a:
      x = x * 2 + 2
      i += 1
    print(str(x))
else:
  x = 3582
  a -= 11
  a %= 1562511 - 11
  i = 0
  while i < a:
    x = (2*x + 2) % mod
    i += 1
  print(str(x + mod)[1:])
  
